sys/riscv/riscv/locore.S

```diff
@@ -235,6 +280,7 @@ _start:
 	/* Page tables END */
 
 	/* Setup supervisor trap vector */
+2:
 #ifdef __CHERI_PURE_CAPABILITY__
 	cllc	ct0, va
 	li	t1, KERNBASE
```

Comment:
GitHub is a load of garbage so I can't add the review notes needed in lines below directly.  I'll try to list them here instead.

The `#ifdef SMP` to store the `boot_hart` for the purecap case needs to be removed.

Need to save 'a0' somewhere.  I would probably save it in `cs2` as an address of ddc, something like:

```
      cspecialr    ct0, ddc
      csetaddr    cs2, a0
```

I don't think we have any way to know the bounds unfortunately, but we could perhaps at least pick something like 2MB as an upper bound.

`cs2` then needs to be written to `RISCV_BOOTPARAMS_MODULEP(csp)` as it is in the non-purecap case.